rfcs/2025-03-16-smp-queues.md

```diff
@@ -0,0 +1,163 @@
+- note that the sender does not authorize the request to retrieve the blob, as it would not increase security unless a different key is used to authorize, and adding a key would increase link size.
+- server session keys with the sender will be `(sk, spk)`, where `sk` is public key shared with the sender during session handshake, and `spk` is the private key known only to the server.
+- this public key `k` will also be combined with server session key `spk` using `dh(k, spk)` to encrypt the response, so that there is no ciphertext in common in sent and received traffic for these blobs. Correlation ID will be used as a nonce for this encryption.
+- having received the blob, the client can now decrypt it using secret_box with `HKDF(pk)`.
+
+Using the same key as ID for the request, and also to additionally encrypt the response allows to use a single key in the link, without increasing the link size.
+
+## Threat model
+
+See [Short links](./2024-06-21-short-links.md) RFC.
+
+## Syntax for short links
+
+The proposed syntax:
+
+```abnf
+shortConnectionRequest = %s"https://" smpServerHost "/" connReqType "#/" linkKey [ "?" param *( "&" param ) ]
+smpServerHost = <hostname> ; RFC1123, RFC5891
+connReqType = %s"i" / %s"invitation" / %s"c" / %s"contact"
+linkKey = <base64url encoded private key data>
+
+param = fingerprintParam / hostsParam
+fingerprintParam = "f=" fingerprint ; server fingerprint
+hostsParam = "h=" <hostname> *( "," <hostname> ) ; additional hostnames, e.g. onion
+```
+
+To have shorter links fingerpring and additional server hostnames do not need to be specified for preconfigured servers, even if they are disabled - they can be used from the client code.
+
+Example link:
+
+```
+https://smp8.simplex.im/c/#abcdefghij0123456789abcdefghij0123456789abc
+```
+
+For the above to work in the browser the servers should serve server pages.
+
+## Questions
+
+SenderId determination:
+- random, as proposed. Pros: no linkability of observed link to queue. Cons: it can be different from the queue, without the accepting party knowing. Potentially, an attack vector?
+- derived from LinkId, and not changeable: Pros: accepting party can verify. Cons: observed link is linkable to the queue.
+
+Possibly, we could split data blobs to mutable and immutable parts, where SenderId would be immutable, so while it preserves constant SenderId it remains unlinkable.
+
+On another hand, having mutable sender ID may allow delegating the reception of messages to chat relay - when initially messages are received by the user, different contact request URI could point to another SMP client of chat relay, similar to how we plan to do with the group links.
```

Comment:
didn't get this.. as in allow that client to subscribe with same key / changed key?

Reply:
The connection request inside the link can be created/controlled by chat relay. That would require some e2e via chat relay, but its parameters could be part of the blob, together with the link.